extensions/native/guest-macro/src/lib.rs

```diff
@@ -0,0 +1,23 @@
+use p3_baby_bear::BabyBear;
+
+use crate::parse_compiler_output::CompiledKernel;
+
+mod parse_compiler_output;
+mod parse_kernel;
+mod transportation;
+const UTILITY_CELL: usize = 0;
+
+#[proc_macro]
+pub fn edsl_kernel(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let input = proc_macro2::TokenStream::from(input);
+
+    let parsed_kernel = parse_kernel::parse_raw_kernel(input);
+    let compiler_output = std::fs::read_to_string(parsed_kernel.file_path.clone()).unwrap();
```

Comment:
not a strong opinion, but we could store the `Vec<u32>` as a raw binary file of `Vec<u8>`. I'm not sure we ever care its readable as a string?